pkg/hub/controllers/placemanifestworkcontroller/placemanifestwork_deploy_reconcile.go

```diff
@@ -18,5 +30,133 @@ type deployReconciler struct {
 
 func (d *deployReconciler) reconcile(ctx context.Context, pw *workapiv1alpha1.PlaceManifestWork) (*workapiv1alpha1.PlaceManifestWork, reconcileState, error) {
 	// TODO the main manifestwork create/update/delete logic, we may also need to manage the status condition for whether manifestwork is created here
-	return pw, reconcileContinue, nil
+	klog.Info("reconciling placeManifestWork ", pw.Name)
+
+	placement, err := d.placementLister.Placements(pw.Namespace).Get(pw.Spec.PlacementRef.Name)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			apimeta.SetStatusCondition(&pw.Status.Conditions, GetPlacementDecisionVerified(workapiv1alpha1.PlacementDecisionNotFound, ""))
+		}
+		return pw, reconcileContinue, fmt.Errorf("Failed get placement %w", err)
+	}
+
+	//klog.Info("my reconciling placement ", placement)
+	errs := []error{}
+	req, err := labels.NewRequirement(PlaceManifestWorkControllerNameLabelKey, selection.In, []string{pw.Name})
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	selector := labels.NewSelector().Add(*req)
+	manifestWorks, err := d.manifestWorkLister.List(selector)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	var existingClusters sets.String
+	for _, mw := range manifestWorks {
+		existingClusters.Insert(mw.Namespace)
+	}
+
+	addedClusters, deletedClusters, err := helper.GetClusters(d.placeDecisionLister, placement, existingClusters)
+	if err != nil {
+		return pw, reconcileContinue, utilerrors.NewAggregate(errs)
+	}
+	//klog.Info("Selected cluters count ", len(added), len(deleted))
+
+	workApplier := workapplier.NewWorkApplierWithTypedClient(d.workClient, d.manifestWorkLister)
+
+	if len(addedClusters) == 0 && len(existingClusters) == 0 {
+		apimeta.SetStatusCondition(&pw.Status.Conditions, GetPlacementDecisionVerified(workapiv1alpha1.PlacementDecisionEmpty, ""))
+		apimeta.SetStatusCondition(&pw.Status.Conditions, GetManifestworkApplied(workapiv1alpha1.PlacementDecisionEmpty, ""))
+	}
+
+	// Delete manifestWork for deleted clusters
+	for cls := range deletedClusters {
+		err = workApplier.Delete(ctx, cls, pw.Name)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	// Create manifestWork for added clusters
+	for cls := range addedClusters {
+		mw, err := CreateManifestWork(pw, cls)
+		if err != nil {
+			return pw, reconcileContinue, nil
+		}
+
+		_, err = workApplier.Apply(ctx, mw)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	// Update created manifestWorks in case there are changes at ManifestWork or PlaceManifestWork
+	for _, mw := range manifestWorks {
+		if deletedClusters.Has(mw.Namespace) {
+			continue
+		}
+
+		// TODO: waiting for the PR https://github.com/open-cluster-management-io/api/pull/200 to be merged
```

Comment:
I do not think you need this.

Reply:
yeah, but applier should update those, right? It is not necessary to treat it differently with create process.